packages/flutter/lib/src/widgets/transitions.dart

```diff
@@ -284,10 +287,25 @@ class ScaleTransition extends AnimatedWidget {
 
   @override
   Widget build(BuildContext context) {
+    // The ImageFilter layer created by setting filterQuality will introduce
+    // a saveLayer call. This is usually worthwhile when animating the layer,
+    // but leaving it in the layer tree before the animation has started or after
+    // it has finished significantly hurts performance.
+    final bool skipfilterQuality;
```

Comment:
A nit. This is a negative name so that we have to do something if it is not set. I tend to prefer positive names that indicate something needs to be done if the flag is set. We're focused on "skipping" the filter quality because that is the improvement we want to see, but this flag is really indicating whether to use the filter quality. Does that make any semantic sense?

So, would "useFilterQuality" and swap all of the true/false/?: logic read better in this code?

Reply:
The exceptional and uncommon condition here is that we are in the process of animating, so the default is to not forward the filterQuality and the condition to call out are the cases when we do want to use it...?

Reply:
Yeah that reads better to me. Updated.